### PR TITLE
EVG-8236: Fix 'My patches' breadcrumb on another user's patch leads to my own patches

### PR DESCRIPTION
**Pull request by SupaJoon**
https://jira.mongodb.org/browse/EVG-8236
These code changes rely on https://github.com/evergreen-ci/evergreen/pull/3816
The first breadcrumb will say "My Patches" and link to your patches page when viewing a Task or Patch that belongs to you. 
The first breadcrumb will be formatted as "{Other person's display name}{ ' | 's} Patches" and link to that person's patches page when viewing that person's Task or Patch. 
The User Patches page and tab title will say "My Patches" when viewing your own patches page.
The User Patches page and tab title will say "{Other person's display name}{ ' | 's} Patches" when viewing another person's patches page.

### CODE REVIEW

File: src/components/Breadcrumb/index.tsx [hunk 1]
```diff
@@ -1,43 +1,49 @@
...
       <Breadcrumb.Item>
         <StyledP1>
           <StyledBreadcrumbLink
             id="bc-my-patches"
-            to={routes.myPatches}
+            to={userPatchesPageLink}
             onClick={() =>
               breadcrumbAnalytics.sendEvent({
                 name: "Click Link",
                 link: "myPatches",
```
Comment (SupaJoon) [inline src/components/Breadcrumb/index.tsx:42]: Do ya'll think we should adjust the analytics for this page to include the author userId?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "spruce",
-  "version": "0.5.1",
+  "version": "0.5.2",
```
Comment (SupaJoon) [inline package.json:3]: Deploying to production this morning bumped the version in my package.json. Do ya'll think I should do this in a separate PR?

File: src/components/Breadcrumb/index.tsx (hunk 1 above)
Comment (john-m-liu) [reply src/components/Breadcrumb/index.tsx:42]: I don't think so, unless we care about seeing when people click on other people's patches vs their own

File: src/hooks/useGetUserPatchesPageTitleAndLink.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  if (loading || error) {
+    return { error, link, loading, title: "Patches" };
+  }
+
+  if (userId === data?.currentUser.userId) {
+    return { error, link, loading, title: "My Patches" };
+  }
+
+  const otherUserDisplayName = data?.otherUser.displayName ?? "";
```
Comment (tgrander) [inline src/hooks/useGetUserPatchesPageTitleAndLink.ts:24]: can we do `data?.otherUser?.displayName` to be extra careful on avoiding type errors in prod?

Comment (SupaJoon) [reply src/hooks/useGetUserPatchesPageTitleAndLink.ts:24]: this is already type safe since the result of the `user` gql query is non-nullable

File: src/hooks/useGetUserPatchesPageTitleAndLink.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  const possessiveModifier =
+    otherUserDisplayName.slice(-1) === "s" ? "'" : "'s";
```
Comment (khelif96) [inline src/hooks/useGetUserPatchesPageTitleAndLink.ts:26]: neat touch!

**Comment by SupaJoon** [comment]
evergreen retry
